src/WebSockets.jl

```diff
@@ -91,6 +91,27 @@ function mask!(bytes::Vector{UInt8}, mask)
     return
 end
 
+function compress(data::T) where T <: AbstractVector{UInt8}
+    compressed = transcode(DeflateCompressor, data)
+    return vcat(compressed, 0x00)
```

Comment:
Looks like you don't need to allocate a full new vector here:
```suggestion
    compressed = transcode(DeflateCompressor, data)
    push!(compressed, 0x00)
    return compressed
```